packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/notifications/RealmDictionaryNotificationsTests.kt

```diff
@@ -0,0 +1,390 @@
+                assertEquals(1, dictionaryChange.map.size)
+                dictionaryChange.changes.let { changes ->
+                    assertEquals(1, changes.size)
+                    assertEquals(dataSet[0].first, changes[0])
+                }
+                assertEquals(0, dictionaryChange.deletions.size)
+            }
+
+            // Assert multiple insertions at once are reported
+            realm.writeBlocking {
+                val queriedContainer = findLatest(container)
+                val queriedDictionary = queriedContainer!!.nullableObjectDictionaryField
+                queriedDictionary.putAll(dataSet.subList(1, dataSet.size))
+            }
+
+            channel.receive().let { dictionaryChange ->
+                assertIs<UpdatedDictionary<*>>(dictionaryChange)
+
+                assertNotNull(dictionaryChange.map)
+                assertEquals(dataSet.size, dictionaryChange.map.size)
+                dictionaryChange.insertions.let { insertions ->
+                    assertEquals(dataSet.size - 1, insertions.size)
+                    dataSet.map { it.first }
+                        .also { keys ->
+                            keys.containsAll(insertions.toList())
+                        }
+                }
+                assertEquals(0, dictionaryChange.deletions.size)
+            }
+
+            // Assert notification on removal of elements
+            realm.writeBlocking {
+                val queriedContainer = findLatest(container)!!
+                val queriedDictionary = queriedContainer.nullableObjectDictionaryField
+                queriedDictionary.remove(dataSet[0].first)
+            }
+
+            channel.receive().let { dictionaryChange ->
+                assertIs<UpdatedDictionary<*>>(dictionaryChange)
+
+                assertNotNull(dictionaryChange.map)
+                assertEquals(dataSet.size - 1, dictionaryChange.map.size)
+                assertEquals(1, dictionaryChange.deletions.size)
+                assertEquals(0, dictionaryChange.insertions.size)
+            }
+
+            // Assert notification on removal of elements via values iterator
+            realm.writeBlocking {
+                val queriedContainer = findLatest(container)!!
+                val queriedDictionary = queriedContainer.nullableObjectDictionaryField
+                val iterator = queriedDictionary.values.iterator()
+                iterator.next()
+                iterator.remove()
+            }
+
+            channel.receive().let { dictionaryChange ->
+                assertIs<UpdatedDictionary<*>>(dictionaryChange)
+
+                assertNotNull(dictionaryChange.map)
+                assertEquals(dataSet.size - 2, dictionaryChange.map.size)
+                assertEquals(1, dictionaryChange.deletions.size)
+                assertEquals(0, dictionaryChange.insertions.size)
+            }
+
+            // Assert notification on removal of elements via entry set iterator
```

Comment:
This seems to be more about testing the value iterator. So I would pull this into its own test case.

Reply:
I think it makes sense to leave it here. Dictionary notifications can be triggered from multiple places: by modifying the entry set, the value collection or the dictionary itself. That is different from other collections. I think having everything under the `asFlow` call proves that the same stream can handle being bombarded from all different change sources just fine.